src/elisp/treemacs-mode.el

```diff
@@ -384,6 +384,8 @@ Will simply return `treemacs--eldoc-msg'."
   (when treemacs-fringe-indicator-mode
     (treemacs--setup-fringe-indicator-mode))
   (hl-line-mode t)
+  ;; Required for skipping hidden buttons
+  (cursor-sensor-mode t)
```

Comment:
`cursor-sensor-functions` requires this.

Reply:
This is supported in Emacs 25. `cursor-intangible-mode` is supposed to do precisely what we want, but for some reason or other it does not.

Reply:
That mode works by hooking into `pre-redisplay-functions`, so it gets called pretty often (at least 3 times just for inserting a character). Would it not be easier just to define our own post-command-hook?

Reply:
Updated this to use post-command. I renamed the default-directory function to `treemacs--post-command` and put the logic there.

Reply:
...And pushed again, to fix setting current dir and eldoc when avoiding invisible buttons.